### DISCUSSION TITLE
I'm having dificulties to use patches

### DISCUSSION THREAD

**Discussion by osvik** [jesseduffield/lazygit#2178] [Q&A]
I couldn't do the `format-patch` and `am`in a repo with Lazygit although it's quite easy with git in the command line.

Is this done by the "view custom patch options"? I keep getting the _"No patch created yet..."_ error. It's confusing to me.

In the command line I do (from the feature branch):
`git format-patch main -o patches`
to create it and:
`git am < patches/*`
to import it.

I'm using:
build source=macports, version=0.35, os=darwin, arch=arm64

**Comment by mark2185** [comment]
Take a look at the [introduction to lazygit video](https://m.youtube.com/watch?v=CPLdltN7wgE&t=442s), I think this part explains it.

**Reply by osvik** [reply]
Thank you. I was actually looking on how to accept contributions by importing a patch instead of using Github's forks and pull requests.

**Reply by mark2185** [reply]
Well, that might be something feature worthy - applying patches through `lazygit`. @jesseduffield ?

**Reply by jesseduffield** [reply]
Agreed that is indeed feature worthy. Is there a UI flow you have in mind @osvik ?

**Reply by mark2185** [reply]
Maybe we could just let the user pick a file to load a patch from when pressing `<C-P>` if there is no patch? I.e. a `menu` instead of an `error`. And then... then I'm not sure. In any case, I think it should synergy well with the current `Custom patch` functionality.

**Reply by jesseduffield** [reply]
I agree, if we can get it to work well with the existing stuff that would be best.

**Comment by osvik** [comment]
One use case of this feature is when users develop code in a staging or shared development server. This happens when the user doesn't has a local environment. Staging servers often don't have push permissions to Github, but the user can create a patches folder, zip it and send it to a maintainer to merge it into the repo.

And although many users use Github, Bitbucket or Gitlab, we should be able to develop without one.

There's workflows that integrate patches with email, but this would be another step.

**As a developer I:**
- Create a feature branch
- Develop and commit in that feature branch
- Create patches against the base branch in a folder `git format-patch main -o patchesfolder`
- Zip the patches folder and send it to a user with push permissions

**The user with push permissions:**
- Unzips the patches folder
- Creates a new branch from the same base commit
- Uses `git am < patchesfolder/*`
- Inspects the code and merges the feature branch into another branch

I know many people are using workflows that depend on CI, Github and other stuff, but I still think this can be useful for some use cases.
